### PR TITLE
Make validation set optional

### PR DESCRIPTION
**Pull request by gietema**

### CODE REVIEW

File: tfimm/train/trainer.py [hunk 1]
```diff
@@ -172,7 +173,8 @@ def train(self):
 
                 it += 1
 
-            self.validation()
+            if self.val_ds:
```
Comment (martinsbruveris) [inline tfimm/train/trainer.py:176]: I would move this check into `self.validation()`. That way you need to check whether `val_ds is None` in only one place.

File: tfimm/train/train.py [hunk 2]
```diff
@@ -25,8 +25,8 @@ class ExperimentConfig:
     # Data
     train_dataset: Any
     train_dataset_class: str
-    val_dataset: Any
-    val_dataset_class: str
+    val_dataset: Optional[Any] = None
+    val_dataset_class: Optional[str] = None
```
Comment (martinsbruveris) [inline tfimm/train/train.py:29]: The difficulty with `Optional` and config classes is that I am using the typehint as type information for parsing command line arguments. Which is why types for config classes are fairly strict. We can still have the val dataset optional by passing the empty string as `val_dataset_class`.

And for `Any` you don't need `Optional`. `Any` covers everything already.

File: tfimm/train/train.py [hunk 3]
```diff
@@ -80,8 +80,9 @@ def run(cfg: ExperimentConfig, parse_args: bool = False):
 
     # Construct datasets
     train_ds = get_class(cfg.train_dataset_class)(cfg=cfg.train_dataset)
-    # TODO: Make val dataset optional
-    val_ds = get_class(cfg.val_dataset_class)(cfg=cfg.val_dataset)
+    val_ds = None
+    if cfg.val_dataset and cfg.val_dataset_class:
```
Comment (martinsbruveris) [inline tfimm/train/train.py:84]: I would make the check here `if cfg.val_dataset_class:`, the assumption being, if we have a value for `val_dataset_class` in `cfg`, we expect correct config to be present in the key `val_dataset`. We don't want to silently skip validation if `val_dataset` doesn't contain a valid config.

File: tfimm/train/train.py (hunk 2 above)
Comment (gietema) [reply tfimm/train/train.py]: Thanks for the comments. We still need to provide default values, so I now have
`val_dataset: Any = None` and `val_dataset_class: str = ""`

Comment (martinsbruveris) [reply tfimm/train/train.py]: Yes, that makes sense. We also need to adjust config parsing to allow for empty strings for `_class` fields. I will do that now in the `feat/config` branch.

Thanks for the PR.
